addons/web/static/src/legacy/scss/ui.scss

```diff
@@ -66,6 +66,48 @@
     animation: catchAttention 200ms ease 0s infinite normal;
 }
 
+// ----------------------------------------------------------------------------
+// Render a "tree-view" design on a set of vertical elements
+
+// @param {number} --treeEntry-padding-h - Entry's horizzontal padding.
+// @param {number} --treeEntry-padding-v - Entry's vertical padding.
+// @param {number} --treeEntry--before-top - Vertical-line top position.
+// @param {string} --treeEntry--after-display - Horizzontal-line display mode.
+// @param {color}  --treeEntry--beforeAfter-color - Lines color
+// @param {color}  --treeEntry--beforeAfter-left - Lines left position
+// ----------------------------------------------------------------------------
+.o_treeEntry {
```

Comment:
First of all, I strongly dislike the mismatch of naming conventions ; whichever we choose should be applied globally.

In this case (ie. the class name), it looks more like:
- `o_` is a legacy, snake-case, prefix
- `treeEntry` is a js-like class/component instance name, camel-case... but actually no js is involved in this case :person_shrugging: 

In the same way, for (S)CSS variables, I do understand the BEM-like separators but I would suggest using more descriptive names for `:before`/`:after` pseudo-elements.

So, after some discussion in our team and/or the Framework, my suggestion is the following:
- `o` prefix for Odoo classes/variables/ids...
- lowercase names in a snake/kebab case style
- same separator for the whole name (`-` like in the recent framework merges, `_` like in the documentation).
- descriptive name instead of implementation-bound ones (can be tricky sometimes).

```suggestion
// @param {number} --o-tree-entry-padding-h - Entry's horizontal padding.
// @param {number} --o-tree-entry-padding-v - Entry's vertical padding.
// @param {number} --o-tree-entry-line-v-top - Vertical-line top position.
// @param {string} --o-tree-entry-line-h-display - Horizontal-line display mode.
// @param {color}  --o-tree-entry-line-color - Lines color
// @param {color}  --o-tree-entry-line-left - Lines left position
// ----------------------------------------------------------------------------
.o_tree_entry {
```

Reply:
meh.. not sure about that :/

Since we moved to a component-based system we have to create variables specifically for components.
These variables should reflect the component name.

###### `treeEntry-title-color--hover`

 works better than
###### `tree-entry-title-color-hover`
In the first case **I can immediately spot the component name**, the sub-element affected, the property  and the interaction state. The second case looks random to me and, again, the component name can be easily mislead. 
I do agree on avoid mismatches and the proposed changes look cleaner indeed...  but are way less effective IMO.

It's not my intention to re-open the discussion but, since I was not there when it happened, I'm just curios to understand if you thought about these aspects.

Also, I'd like to have @qsm-odoo opinion, we should have the same conventions between frontend & backend (for context, [our initial Frankenstein proposal](https://github.com/odoo/odoo/pull/87687#discussion_r909795845)   , definitively not clean but gives an idea of what we need)

Reply:
FYI @odoo/rd-framework-js